src/Xamarin.Android.Build.Tasks/Tasks/InstallApkSet.cs

```diff
@@ -28,8 +30,11 @@ internal override CommandLineBuilder GetCommandLineBuilder ()
 			// --modules: List of modules to be installed, or "_ALL_" for all modules.
 			// Defaults to modules installed during first install, i.e. not on-demand.
 			// Xamarin.Android won't support on-demand modules yet.
-			cmd.AppendSwitchIfNotNull ("--modules ", "_ALL_");
-			
+			if (Modules == null)
+				cmd.AppendSwitchIfNotNull ("--modules ", "_ALL_");
+			else
+				cmd.AppendSwitchIfNotNull ("--modules", string.Join ("\",\"", Modules));
```

Comment:
I don't think this would put the leading & ending `"`, so you would get:
```
--modules foo","bar
```

Reply:
Or does `AppendSwitchIfNotNull` do that for you?

Reply:
Also this might be needed:
```suggestion
				cmd.AppendSwitchIfNotNull ("--modules ", string.Join ("\",\"", Modules));
```